crates/ty_python_semantic/src/semantic_index/scope.rs

```diff
@@ -14,7 +14,13 @@ use crate::{
 };
 
 /// A cross-module identifier of a scope that can be used as a salsa query parameter.
-#[salsa::tracked(debug, heap_size=ruff_memory_usage::heap_size)]
+///
+/// This type is an interned struct as opposed to a tracked struct as it is the input of
+/// the `infer_scope_types` query, which is the root query of the persistent cache.
+///
+/// Additionally, this type disables garbage collection, as garbage collectable interned values are volatile,
+/// and would cause `infer_scope_types` to have a dependency on every single `ScopeId` in the file.
```

Comment:
Let's add a `TODO` to consider reverting to a tracked struct once supported by the persistent caching backend.

Reply:
Hmm, I'm not sure how this would work without persisting the `semantic_index` query.